### PR TITLE
Build change info additional features

### PR DESCRIPTION
**Pull request by stencelm**
Modified generate_build_change_info.py to take input of two builds instead of one (and decreasing the second build number by 1). Added last_successful.py script which queries the database looking for the last successful periodic-nightly build before the current one.

### CODE REVIEW

File: tungsten_ci_utils/generate_build_change_info/last_successful.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    return data
+
+def close_db_exit_err(db):
+    if db is not None:
+        db.close()
+    sys.exit(1)
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--credentials-json", action="append")
```
Comment (jluk-codi) [inline tungsten_ci_utils/generate_build_change_info/last_successful.py]: In which cases do we need multiple "credentials-json" options?

File: tungsten_ci_utils/generate_build_change_info/last_successful.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    parser.add_argument("branch")
+    parser.add_argument("build_number")
+    args = parser.parse_args()
+    branch = str(args.branch)
+    credentials_json = args.credentials_json[0]
+
+    build_number = og_build_number = int(args.build_number)
```
Comment (jluk-codi) [inline tungsten_ci_utils/generate_build_change_info/last_successful.py]: Set "int" type of the "build_number" argument to get automatic validation and conversion.

File: tungsten_ci_utils/generate_build_change_info/last_successful.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        '%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    console = logging.StreamHandler()
+    console.setLevel(logging.DEBUG)
+    console.setFormatter(formatter)
+    log.addHandler(console)
+
+set_logging()
+
+def get_json_data(file):
+    json_file = open(file).read()
```
Comment (jluk-codi) [inline tungsten_ci_utils/generate_build_change_info/last_successful.py]: Close the file after reading or use the "with" statement to have it closed automatically:
http://cmdlinetips.com/2016/01/opening-a-file-in-python-using-with-statement/

File: tungsten_ci_utils/generate_build_change_info/last_successful.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        cur = db.cursor()
+        log.debug('connected to db')
+    except:
+        log.error('database connection error')
+        sys.exit(1)
+
+    while build_number > 0:
+        query = """
+        SELECT result FROM zuul_buildset WHERE id IN 
+        (SELECT buildset_id FROM zuul_build WHERE log_url LIKE CONCAT('%%periodic-nightly%%/', %s, '/%s%%'))
```
Comment (jluk-codi) [inline tungsten_ci_utils/generate_build_change_info/last_successful.py]: Try matching branch and pipeline explicitly using SQL column names (will be more readable in the future, compared to matching the log URL string).

Comment (kravvcuCL) [reply tungsten_ci_utils/generate_build_change_info/last_successful.py]: Also, you can directly add a condition "and result == 'SUCCESS'" here, unless you wanted to iterate and log the failed builds.

File: tungsten_ci_utils/generate_build_change_info/last_successful.py (hunk 1 above)
Comment (stencelm) [reply tungsten_ci_utils/generate_build_change_info/last_successful.py]: We don't. Fixed.

**Comment by stencelm** [comment]
This ended up with a lot of commits so I'm going to squash them before merging.

**Review by kravvcuCL — APPROVED** [review]

**Pull request merged**